From 9e137e2aa2ca086421616ae05d288da4eefc26ea Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 20 Dec 2021 06:08:55 +0200
Subject: [PATCH] Revert "[SW-63186] gaudi2: security: set axprot for pdma"

As per alignment with LKD, MSTR_IF's OVRD register should not be used
to set to "USER level".

This is because sub-blocks which should be secure won’t be able to work,
as MSTR_IF's OVRD bit takes precedence over the engine's internal PROT
configuration.

This reverts commit d6257eb008ed4d7b7d308816e9b0901a46b394ec.

Change-Id: I20ab19e8cea4cf884cc9945fda27abe8eca47b96
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 5393328..e7ddf0f 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -43,8 +43,6 @@
 #include <dcore0_mme_wb0_mstr_if_core_hbw_regs.h>
 #include <dcore0_mme_ctrl_mstr_if_core_hbw_regs.h>
 #include <dcore0_mme_ctrl_mstr_if_core_lbw_regs.h>
-#include <pdma0_mstr_if_core_hbw_regs.h>
-#include <pdma0_mstr_if_core_lbw_regs.h>
 
 /*
  * instances_num : Number of instances of RR to program.
@@ -206,7 +204,6 @@
 enum preboot_sec_axprot_regs {
 	AXPROT_ROT,
 	AXPROT_MME,
-	AXPROT_PDMA,
 };
 
 #define DCORE_DIST	0x200000
@@ -927,10 +924,6 @@
 #define MME_INST_NUM		NUM_OF_DCORES
 #define MME_INST_DIST		DCORE_DIST
 #define MME_CRTL_RTR_INST_NUM	1
-#define PDMA_RTR_INST_NUM	1
-#define PDMA_RTR_INST_DIST	0
-#define PDMA_INST_NUM		2
-#define PDMA_INST_DIST		0x10000
 
 static struct axprot_regs axprot_reg[] = {
 	[AXPROT_ROT] = {
@@ -948,14 +941,6 @@
 		.dcore_num = MME_INST_NUM,
 		.inst_dist = MME_SBTE_RTR_INST_DIST,
 		.inst_num = MME_SBTE_RTR_INST_NUM,
-	},
-	[AXPROT_PDMA] = {
-		.base_reg = mmPDMA0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
-		.value = AXPROT_USER_VAL,
-		.dcore_dist = PDMA_INST_DIST,
-		.dcore_num = PDMA_INST_NUM,
-		.inst_dist = PDMA_RTR_INST_DIST,
-		.inst_num = PDMA_RTR_INST_NUM,
 	},
 };
 
@@ -1031,19 +1016,6 @@
 	preboot_axprot_regs_init(AXPROT_MME);
 }
 
-static void preboot_secure_pdma(void)
-{
-
-	/* Configure pdma to user level */
-	/* HBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_PDMA);
-
-	axprot_reg[AXPROT_PDMA].base_reg = mmPDMA0_MSTR_IF_CORE_LBW_AXPROT_OVRD;
-
-	/* LBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_PDMA);
-}
-
 void preboot_security_init(void)
 {
 	preboot_secure_pcie();
@@ -1057,7 +1029,6 @@
 		preboot_secure_arc_farm();
 		preboot_secure_trace();
 		preboot_secure_mme();
-		preboot_secure_pdma();
 		preboot_secure_lbw();
 		preboot_secure_hbw();
 		preboot_puts("Security enabled.\n");
